### DISCUSSION TITLE
Invalid JSON - How to log errors as array of objects in JSON?

### DISCUSSION THREAD

**Discussion by burakt91** [RehanSaeed/Serilog.Exceptions#227] [Q&A]
In my output file, errors logging as this:
```
{"Timestamp":"2020-08-11T15:33:45.8624993+03:00","Level":"Error".....}
{"Timestamp":"2020-08-11T15:39:41.3336583+03:00","Level":"Error".....}
.
.
.
```
But this is not a valid JSON, because it has multiple root elements. It's very difficult to pretty print and read.
![Capture](https://user-images.githubusercontent.com/18482904/90118330-c29eb000-dd60-11ea-9044-21b63037e41c.PNG)
How can i generate a valid JSON (array of objects) like this:
```
[
  {"Timestamp":"2020-08-11T15:33:45.8624993+03:00","Level":"Error".....},
  {"Timestamp":"2020-08-11T15:39:41.3336583+03:00","Level":"Error".....},
  .
  .
  .
]
```
My current logger configuration is this:
```
new LoggerConfiguration()
  .Enrich.WithExceptionDetails()
  .WriteTo.Sink(new RollingFileSink(
    pathFormat: Path.Combine(DIR, "ERROR.txt"),
    textFormatter: new JsonFormatter(renderMessage: true),
    fileSizeLimitBytes: 1000 * 1000 * 20, // 20 MB
    retainedFileCountLimit: 3,
  shared: true))
  .CreateLogger();
```
Relevant issues i inspected on Serilog repository:

- https://github.com/serilog/serilog/issues/105
- https://github.com/serilog/serilog/issues/107

**Comment by RehanSaeed** [comment]
This is more of a Serilog question but I'll answer briefly.

With structured logging, there are various consumers that have been built with this format in mind e.g. Filebeat can be used to pick up the json logs and send them to elastic search. Alternatively, instead of using the file logger, you can send logs directly to a consumer e.g. You can send logs directly to Seq.
